Assets/Scripts/Pet/Pet.cs

```diff
@@ -85,18 +86,37 @@ public override void OnInteract()
         // if the pet is not saved, save the pet
         if (!_isSaved)
         {
+            if (TimeManager.Instance.CurrentGlobalTimeState == TimeState.Present)
+            {
+                GameManager.Instance.gameHUD.optionDescription.text = "This looks like skeleton of some animal that died a long time ago...";
+                GameManager.Instance.gameHUD.PresentInteractionMessageOnly();
+            }
+            else if (TimeManager.Instance.CurrentGlobalTimeState == TimeState.Past)
+            {
+                GameManager.Instance.gameHUD.optionDescription.text = "This small injured Slime is looking at you and want to be helped...";
+                GameManager.Instance.gameHUD.yesButton.onClick.AddListener(SavePet);
+                GameManager.Instance.gameHUD.PresentInteractionMessageAndOptions(saveCost);
+            }
+            
             SavePet();
```

Comment:
Remove this invocation. Otherwise the pet will be saved on any interaction.

Reply:
issue fixed